contracts/finance/andromeda-vesting/src/contract.rs

```diff
@@ -314,21 +319,64 @@ fn execute_delegate(
     });
 
     Ok(Response::new()
+        .add_message(get_set_withdraw_address_msg(sender))
         .add_message(msg)
         .add_attribute("action", "delegate")
         .add_attribute("validator", validator)
         .add_attribute("amount", amount))
 }
 
+fn execute_redelegate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    amount: Option<Uint128>,
+    from: String,
+    to: String,
+) -> Result<Response, ContractError> {
+    let sender = info.sender.to_string();
+    require(
+        ADOContract::default().is_contract_owner(deps.storage, &sender)?,
+        ContractError::Unauthorized {},
+    )?;
+    let config = CONFIG.load(deps.storage)?;
+    let max_amount = get_amount_delegated(
+        &deps.querier,
+        env.contract.address.to_string(),
+        from.clone(),
+    )?;
+    let amount = cmp::min(max_amount, amount.unwrap_or(max_amount));
```

Comment:
@daniel-wehbe This should be noted in the docs. If the `amount` field is larger than the amount delegated it redelgates the minimum of the two.